modules/bmc/ipmi.rb

```diff
@@ -148,6 +153,124 @@ def netmask
         host.bmc.lan.netmask
       end
 
+      # return the SNMP community string of the bmc device
+      def snmp
+        host.bmc.lan.snmp
+      end
+
+      # return the VLAN ID of the bmc device
+      def vlanid
+        host.bmc.lan.vlanid
+      end
+
+      # return IP source of BMC device
+      def ipsrc
+        if host.bmc.lan.dhcp?
+          return "dhcp"
+        else
+          return "static"
+        end
+      end
+
+      # return all LAN details of BMC device
+      def lanprint
+        host.bmc.lan.info
+      end
+
+      # BMC information
+      def info
+        host.bmc.info
+      end
+
+      # BMC GUID information
+      def guid
+        host.bmc.guid
+      end
+
+      # BMC firmware version
+      def version
+        host.bmc.version
+      end
+
+      # BMC reset
+      def reset(type='cold')
+        host.bmc.reset(type)
+      end
+
+      # print all FRU information
+      def frulist
+        frulist = host.fru.list
+        # Detect quirk on ipmitool
+        raise 'Quirk' if frulist.empty?
```

Comment:
I find this hard to read/follow. Can 'quirk' happen with ipmitool only? If so, why not return the error message on line 218 right away?